### PR TITLE
Auto case OCP-40366

### PR DESCRIPTION
**Pull request by xiuwang**
@openshift/devexp-qe  The case passed on my localhost, http://pastebin.test.redhat.com/954310
Please help review, thanks

### CODE REVIEW

File: features/build/buildlogic.feature [hunk 1]
```diff
@@ -458,3 +458,45 @@ Feature: buildlogic.feature
       | sample-pipeline.*Cancelled |
       | sample-pipeline.*Complete  |
     """
+
+  # @author xiuwang@redhat.com
+  # @case_id OCP-40366
+  @admin
+  Scenario: Mirroring built image doesn't degrade scheme2 ,keep consistent SHA's	
+    Given I have a project
+    Given I save a htpasswd registry auth to the :combare_dockercfg clipboard
```
Comment (pruan-rht) [inline features/build/buildlogic.feature:467]: @xiuwang is this a typo?  `:combare_dockercfg`?

Comment (xiuwang) [reply features/build/buildlogic.feature]: @pruan-rht You caught me 👍  I have updated it, please help review again, thanks

**Comment by wsun1** [comment]
Could this be merged soon now?  Or still need any update?

File: features/step_definitions/helper_services.rb [hunk 2]
```diff
@@ -726,3 +726,33 @@
...
 end
+
+Given /^I save a htpasswd registry auth to the#{OPT_SYM} clipboard$/ do |cb_name|
+  cb_name ||= :dockercfg_file
+  ensure_admin_tagged
+  step 'docker config for default image registry is stored to the :default_dockercfg clipboard'
+  step 'I have a registry with htpasswd authentication enabled in my project'
+  step %Q/I run the :create_route_edge client command with:/, table(%{
+      | name    | registry |
+      | service | registry |
```
Comment (liangxia) [inline features/step_definitions/helper_services.rb:737]: I think we always use 2 whitespaces as indention. The same as below.

File: lib/rules/cli/4.5.yaml [hunk 3]
```diff
@@ -507,6 +507,7 @@
     :filter_by_os: --filter-by-os=<value>
     :insecure: --insecure=<value>
     :skip_verification: --skip-verification=<value>
+    :o: --output=<value>
```
Comment (liangxia) [inline lib/rules/cli/4.5.yaml:510]: Always sort the options.

File: features/step_definitions/helper_services.rb [hunk 4]
```diff
@@ -726,3 +726,33 @@
...
+    | name    | registry |
+    | service | registry |
+  })
+  step 'the step should succeed'
+  step 'I save the hostname of route "registry" in project "<%= project.name %>" to the :custom_registry clipboard'
+  step %Q/a 5 characters random string of type :dns is stored into the :dockercfg_name clipboard/
+  cb[cb_name] ="/tmp/#{cb.dockercfg_name}"
+
+  cb.new_cfg={
+        "#{cb.custom_registry}" => {
```
Comment (xiuwang) [inline features/step_definitions/helper_services.rb:745]: @liangxia  this is a docker auth json file, it has fixed format

**Pull request merged**